src/main/java/org/opensearch/performanceanalyzer/commons/os/ThreadCPU.java

```diff
@@ -0,0 +1,245 @@
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG, // 30
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.ULONG,
+        SchemaFileParser.FieldTypes.INT,
+        SchemaFileParser.FieldTypes.INT,
+        SchemaFileParser.FieldTypes.INT, // 40
+        SchemaFileParser.FieldTypes.INT,
+        SchemaFileParser.FieldTypes.INT,
+        SchemaFileParser.FieldTypes.INT,
+        SchemaFileParser.FieldTypes.INT
+    };
+
+    private ThreadCPU() {
+        try {
+            pid = OSGlobals.getPid();
+            scClkTck = OSGlobals.getScClkTck();
+            tids = OSGlobals.getTids();
+        } catch (Exception e) {
+            LOGGER.error(
+                    (Supplier<?>)
+                            () ->
+                                    new ParameterizedMessage(
+                                            "Error In Initializing ThreadCPU: {}", e.toString()),
+                    e);
+        }
+    }
+
+    public synchronized void addSample() {
+        tids = OSGlobals.getTids();
+
+        oldtidKVMap.clear();
+        oldtidKVMap.putAll(tidKVMap);
+
+        tidKVMap.clear();
+        oldkvTimestamp = kvTimestamp;
+        kvTimestamp = System.currentTimeMillis();
+        for (String tid : tids) {
+            Map<String, Object> sample =
+                    // (new SchemaFileParser("/proc/"+tid+"/stat",
+                    (new SchemaFileParser(
+                                    "/proc/" + pid + "/task/" + tid + "/stat",
+                                    statKeys,
+                                    statTypes,
+                                    true))
+                            .parse();
+            tidKVMap.put(tid, sample);
+        }
+
+        calculateCPUDetails();
+        calculatePagingActivity();
+    }
+
+    public synchronized void addSample(String threadId) {
+        tids = OSGlobals.getTids();
+
+        oldtidKVMap.clear();
+        oldtidKVMap.putAll(tidKVMap);
+
+        tidKVMap.clear();
+        oldkvTimestamp = kvTimestamp;
+        kvTimestamp = System.currentTimeMillis();
+        Map<String, Object> sample =
+                // (new SchemaFileParser("/proc/"+tid+"/stat",
+                (new SchemaFileParser(
+                                "/proc/" + pid + "/task/" + threadId + "/stat",
+                                statKeys,
+                                statTypes,
+                                true))
+                        .parse();
+        tidKVMap.put(threadId, sample);
+
+        calculateCPUDetails();
+        calculatePagingActivity();
+    }
```

Comment:
Why do we need repeated logic here? Update `public synchronized void addSample() {` for parameters with default set as All thread.